### PR TITLE
envoy_reloadable_features_delta_xds_subscription_state_tracking_fix deprecation

### PR DESCRIPTION
**Pull request by DiazAlan**
Commit Message: Removing legacy behavior and runtime guard flag
Additional Description: As per issue #26901 This behavior has been in place for over 6 months and it's time to clean up the flag. PR [23343](https://github.com/envoyproxy/envoy/pull/23343) implemented said change
Risk Level: Low
Testing: Removed legacy behavior from tests
Release Notes: Added note to changelog for removal of flag

### CODE REVIEW

File: source/common/config/new_delta_subscription_state.cc [hunk 1]
```diff
@@ -240,8 +222,7 @@ void NewDeltaSubscriptionState::handleGoodResponse(
                                           message.removed_resources());
   }
 
-  if (Runtime::runtimeFeatureEnabled(
-          "envoy.reloadable_features.delta_xds_subscription_state_tracking_fix")) {
+  {
```
Comment (adisuissa) [inline source/common/config/new_delta_subscription_state.cc:225]: nit: I don't think the inner scope (curly brackets) is needed anymore.

File: source/common/config/xds_mux/delta_subscription_state.cc [hunk 2]
```diff
@@ -211,8 +192,7 @@ void DeltaSubscriptionState::handleGoodResponse(
                                           message.removed_resources());
   }
 
-  if (Runtime::runtimeFeatureEnabled(
-          "envoy.reloadable_features.delta_xds_subscription_state_tracking_fix")) {
+  {
```
Comment (adisuissa) [inline source/common/config/xds_mux/delta_subscription_state.cc:195]: same as above.

**Review by adisuissa — COMMENTED** [review]
Thanks!
Minor comments but otherwise LGTM!

File: source/common/config/xds_mux/delta_subscription_state.cc (hunk 2 above)
Comment (DiazAlan) [reply source/common/config/xds_mux/delta_subscription_state.cc]: Done

File: source/common/config/new_delta_subscription_state.cc (hunk 1 above)
Comment (DiazAlan) [reply source/common/config/new_delta_subscription_state.cc]: Done

File: changelogs/current.yaml [hunk 3]
```diff
@@ -55,6 +55,12 @@ minor_behavior_changes:
     the cookie is still accepted, but is planned to be obsoleted in the future.
     This behavior change can be reverted by setting
     ``envoy.reloadable_features.stateful_session_encode_ttl_in_cookie`` to ``false``.
+- area: config
```
[the remark this answers is not in the document]
Comment (DiazAlan) [reply changelogs/current.yaml]: Ah I didn't see there was a separate category. Fixed

File: source/common/config/new_delta_subscription_state.cc [hunk 4]
```diff
@@ -240,21 +222,18 @@ void NewDeltaSubscriptionState::handleGoodResponse(
                                           message.removed_resources());
   }
 
-  if (Runtime::runtimeFeatureEnabled(
-          "envoy.reloadable_features.delta_xds_subscription_state_tracking_fix")) {
-    const auto scoped_update = ttl_.scopedTtlUpdate();
-    if (requested_resource_state_.contains(Wildcard)) {
-      for (const auto& resource : message.resources()) {
+  const auto scoped_update = ttl_.scopedTtlUpdate();
```
Comment (adisuissa) [inline source/common/config/new_delta_subscription_state.cc:225]: Sorry, made a mistake in the previous comment.
The scope needs to be kept to ensure `scoped_update` is released at the end end of this block.

Comment (DiazAlan) [reply source/common/config/new_delta_subscription_state.cc]: No problem! Fixed it already!

**Review by adisuissa — DISMISSED** [review]
Thanks! LGTM!

/assign-from @envoyproxy/senior-maintainers

**Review by ggreenway — DISMISSED** [review]

**Comment by ggreenway** [comment]
Merge conflicts need resolution

/wait

**Review by ggreenway — APPROVED** [review]

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**